src/services/telemetryService.ts

```diff
@@ -0,0 +1,131 @@
+import * as vscode from 'vscode';
+import Analytics from 'analytics-node';
+import { TelemetryEvent } from '../interfaces/telemetryEvent';
+import { Reporter } from '../reporter';
+import { Logger } from '../utils/logger';
+import { SegmentInitializer } from '../utils/segmentInitializer';
+import { TelemetryEventQueue } from '../utils/telemetryEventQueue';
+
+/* 
+  OPT_IN_STATUS == "true" if user Agreed
+  false when denied for telemetry data collection     
+*/
+const OPT_IN_STATUS = 'redhat.telemetry.optInRequested';
+
+export namespace TelemetryService {
+  let analyticsObject: Analytics | undefined;
+
+  /* 
+    provides subscription to custom segment key.
+    fallback to default segment key if no key provided via API
+  */
+  export function subscribeTelemetryService(
+    clientExtensionName: string
+  ): boolean {
+    Reporter.setClientExtensionName(clientExtensionName);
```

Comment:
Isn't Reporter singleton? How this would work when more than one redhat extension routes events via this extension?

Reply:
@arajkumar each time the adopter extension creates an instance of vscode-commons, so will not encounter any issue here.